File: DTMS365-AppSource/DynamicsTMS365/Models/RateRequestOrder.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace DynamicsTMS365.Models
{
    public class RateRequestOrder
    {
        public int ID { get; set; }
        public string ShipKey { get; set; }
        public string BookConsPrefix { get; set; }
        public string ShipDate { get; set; }
        public string DeliveryDate { get; set; }
        public int BookCustCompControl { get; set; }
        public string CompName { get; set; }
        public int CompNumber { get; set; }
        public string CompAlphaCode { get; set; }
        public int BookCarrierControl { get; set; }
        public string CarrierName { get; set; }
        public int CarrierNumber { get; set; }
        public string CarrierAlphaCode { get; set; }
        public int TotalCases { get; set; }
        public double TotalWgt { get; set; }
        public double TotalPL { get; set; }
        public int TotalCube { get; set; }
        public int TotalStops { get; set; }
        public RateRequestStop Pickup { get; set; }
        public RateRequestStop[] Stops { get; set; }
        public string[] Accessorials { get; set; } //LVV ADD
              
    }

    public class RateRequestStop
    {
        public int ID { get; set; }
        public int ParentID { get; set; }
        public int BookControl { get; set; }
        public string BookProNumber { get; set; }
        public int StopIndex { get; set; }
        public string BookCarrOrderNumber { get; set; }
        public int CompControl { get; set; }
        public string CompName { get; set; }
        public string CompAddress1 { get; set; }
        public string CompAddress2 { get; set; }
        public string CompAddress3 { get; set; }
        public string CompCity { get; set; }
        public string CompState { get; set; }
        public string CompCountry { get; set; }
        public string CompPostalCode { get; set; }
        public bool IsPickup { get; set; }
        public int StopNumber { get; set; }
        public int TotalCases { get; set; }
        public double TotalWgt { get; set; }
        public double TotalPL { get; set; }
        public int TotalCube { get; set; }
        public string LoadDate { get; set; }
        public string SHID { get; set; }
        public string RequiredDate { get; set; }
        public RateRequestItem[] Items { get; set; }
    }

    public class RateRequestItem
    {
        public int ID { get; set; }
        public int ParentID { get; set; }
        public int LoadID { get; set; }
        public int ItemStopIndex { get; set; }
        public int ItemControl { get; set; }
        public string ItemNumber { get; set; }
        public decimal Weight { get; set; }
        //"lbs"
        public string WeightUnit { get; set; }
        //70
        public string FreightClass { get; set; }
        public int PalletCount { get; set; }
        private int _numPieces;
        public int NumPieces {
            get{
                int.TryParse(this.Quantity, out _numPieces);
                return _numPieces;
            }

            set {
                _numPieces = value;               
            }
        }
        public string Description { get; set; }
        public string Quantity { get; set; }
        public string HazmatId { get; set; }
        public string Code { get; set; }
        public string HazmatClass { get; set; }
        public bool IsHazmat { get; set; }
        public string Pieces { get; set; }
        public string PackageType { get; set; }
        public decimal Length { get; set; }
        public decimal Width { get; set; }
        public decimal Height { get; set; }
        public string Density { get; set; }
        public string NMFCItem { get; set; }
        public string NMFCSub { get; set; }
        public bool Stackable { get; set; }
    }

}

File: NGL.DataAssemblies/Ngl.FreightMaster.Data/Models/RateRequestOrder.vb
Imports Microsoft.VisualBasic


Public Class RateRequestOrder

    Private _RRControl As Integer
    Public Property RRControl() As Integer
        Get
            Return _RRControl
        End Get
        Set
            _RRControl = Value
        End Set
    End Property

    Private _RRUserSecurityControl As Integer
    Public Property RRUserSecurityControl() As Integer
        Get
            Return _RRUserSecurityControl
        End Get
        Set
            _RRUserSecurityControl = Value
        End Set
    End Property

    Private _RRBookSHID As String
    Public Property RRBookSHID() As String
        Get
            Return left(_RRBookSHID, 50)
        End Get
        Set
            _RRBookSHID = left(Value, 50)
        End Set
    End Property


    Private _BookConsPrefix As String
    Public Property BookConsPrefix() As String
        Get
            Return left(_BookConsPrefix, 20)
        End Get
        Set
            _BookConsPrefix = left(Value, 20)
        End Set
    End Property

    Private _ShipDate As String
    Public Property ShipDate() As String
        Get
            Return _ShipDate
        End Get
        Set
            _ShipDate = Value
        End Set
    End Property

    Private _DeliveryDate As String
    Public Property DeliveryDate() As String
        Get
            Return _DeliveryDate
        End Get
        Set
            _DeliveryDate = Value
        End Set
    End Property

    Private _BookCustCompControl As Integer
    Public Property BookCustCompControl() As Integer
        Get
            Return _BookCustCompControl
        End Get
        Set
            _BookCustCompControl = Value
        End Set
    End Property

    Private _CompName As String
    Public Property CompName() As String
        Get
            Return left(_CompName, 50)
        End Get
        Set
            _CompName = left(Value, 50)
        End Set
    End Property

    Private _CompNumber As Integer
    Public Property CompNumber() As Integer
        Get
            Return _CompNumber
        End Get
        Set
            _CompNumber = Value
        End Set
    End Property


    Private _CompAlphaCode As String
    Public Property CompAlphaCode() As String
        Get
            Return left(_CompAlphaCode, 50)
        End Get
        Set
            _CompAlphaCode = left(Value, 50)
        End Set
    End Property

    Private _BookCarrierControl As Integer
    Public Property BookCarrierControl() As Integer
        Get
            Return _BookCarrierControl
        End Get
        Set
            _BookCarrierControl = Value
        End Set
    End Property

    Private _CarrierName As String
    Public Property CarrierName() As String
        Get
            Return left(_CarrierName, 50)
        End Get
        Set
            _CarrierName = left(Value, 50)
        End Set
    End Property

    Private _CarrierNumber As Integer
    Public Property CarrierNumber() As Integer
        Get
            Return _CarrierNumber
        End Get
        Set
            _CarrierNumber = Value
        End Set
    End Property

    Private _CarrierAlphaCode As String
    Public Property CarrierAlphaCode() As String
        Get
            Return left(_CarrierAlphaCode, 50)
        End Get
        Set
            _CarrierAlphaCode = left(Value, 50)
        End Set
    End Property

    Private _TotalCases As Integer
    Public Property TotalCases() As Integer
        Get
            Return _TotalCases
        End Get
        Set
            _TotalCases = Value
        End Set
    End Property

    Private _TotalWgt As Double
    Public Property TotalWgt() As Double
        Get
            Return _TotalWgt
        End Get
        Set
            _TotalWgt = Value
        End Set
    End Property

    Private _TotalPL As Double
    Public Property TotalPL() As Double
        Get
            Return _TotalPL
        End Get
        Set
            _TotalPL = Value
        End Set
    End Property

    Private _TotalCube As Integer
    Public Property TotalCube() As Integer
        Get
            Return _TotalCube
        End Get
        Set
            _TotalCube = Value
        End Set
    End Property

    Private _TotalStops As Integer
    Public Property TotalStops() As Integer
        Get
            Return _TotalStops
        End Get
        Set
            _TotalStops = Value
        End Set
    End Property

    Private _Pickups As RateRequestStop()
    Public Property Pickups() As RateRequestStop()
        Get
            Return _Pickups
        End Get
        Set
            _Pickups = Value
        End Set
    End Property

    Private _Stops As RateRequestStop()
    Public Property Stops() As RateRequestStop()
        Get
            Return _Stops
        End Get
        Set
            _Stops = Value
        End Set
    End Property

    Private _ModDate As String
    Public Property ModDate() As String
        Get
            Return _ModDate
        End Get
        Set
            _ModDate = Value
        End Set
    End Property

End Class


